src/serialization.rs

```diff
@@ -10,7 +10,7 @@ use header::Header;
 
 /// The return type of a successful call to decode
 #[derive(Debug)]
-pub struct TokenData<T: Deserialize> {
+pub struct TokenData<T: DeserializeOwned> {
```

Comment:
This bound should be removed: just `pub struct TokenData<T> { ... }`. See https://github.com/Manishearth/rust-clippy/issues/1689.